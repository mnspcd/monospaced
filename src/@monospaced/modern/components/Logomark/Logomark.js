import PropTypes from "prop-types";
import React from "react";

import { mdrnColor as colors } from "../../tokens/color";

import "./Logomark.css";

/**
 * Use `Logomark` to display the Monospaced brand logomark.
 */
const Logomark = ({
  backgroundColor = null,
  color = "currentColor",
  padding = null,
  secondColor = false,
  title = null,
  width = null,
}) => {
  return (
    <span
      className="Logomark"
      style={{ backgroundColor: colors[backgroundColor], width }}
    >
      <span className="Logomark-inner">
        <svg
          className="Logomark-svg"
          role={title && "img"}
          style={{ padding }}
          viewBox="0 0 2048 2048"
          xmlns="http://www.w3.org/2000/svg"
          aria-hidden={!title || null}
          aria-label={title}
        >
          <path
            d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
            fill={colors[color] || "currentColor"}
          />
          {secondColor && (
            <path
              d="M767.766 1640.847h-128.51V406.846H764.35l258.78 615.283 258.8-615.283h126.822v1234h-128.54V701.67h-3.428l-253.655 604.97-251.925-604.97-3.448 3.385v935.793z"
              fill={
                typeof secondColor === "string"
                  ? colors[secondColor]
                  : colors["color-white"]
              }
            />
          )}
        </svg>
      </span>
    </span>
  );
};

Logomark.colors = Object.keys(colors).filter(
  colorKey => !colorKey.includes("-rgb"),
);

Logomark.propTypes = {
  /**
   * Background color of the Logomark.
   */
  backgroundColor: PropTypes.oneOf(Logomark.colors),

  /**
   * Fill color of the Logomark SVG.
   */
  color: PropTypes.oneOfType([
    PropTypes.oneOf([`currentColor`]),
    PropTypes.oneOf(Logomark.colors),
  ]),

  /**
   * A valid CSS padding value for the Logomark SVG.
   */
  padding: PropTypes.string,

  /**
   * Fill color of the “M” path in the Logomark SVG. A value of true adds a
   * white "M" path.
   */
  secondColor: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.oneOf(Logomark.colors),
  ]),

  /**
   * Title of the Logomark. Can be used to provide an accessible label text.
   */
  title: PropTypes.string,

  /**
   * A valid CSS width value for the Logomark.
   */
  width: PropTypes.string,
};

export default Logomark;
