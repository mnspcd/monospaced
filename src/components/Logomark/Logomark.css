.Logomark {
  display: block;
  max-width: 100%;
}

.Logomark-inner {
  display: block;
  height: 0;
  padding-bottom: 100%;
  position: relative;
  width: 100%;
}

.Logomark-svg {
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  width: 100%;
}
