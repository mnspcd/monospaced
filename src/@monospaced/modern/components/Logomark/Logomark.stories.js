import { select, text } from "@storybook/addon-knobs";
import React from "react";

import Group from "../Group";

import Logomark from "./Logomark";

export default {
  component: Logomark,
  title: "Components/Logomark",
};

export const basic = () => <Logomark title="Monospaced" width="6em" />;

export const blue = () => (
  <Group>
    <Group.Item>
      <Logomark color="color-azure-4" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-blue-7" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-blue-9" width="6em" />
    </Group.Item>
  </Group>
);

export const green = () => (
  <Group>
    <Group.Item>
      <Logomark color="color-green-4" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-green-7" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-green-7" secondColor="color-green-3" width="6em" />
    </Group.Item>
  </Group>
);

export const red = () => (
  <Group>
    <Group.Item>
      <Logomark color="color-red-6" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-red-8" width="6em" />
    </Group.Item>
    <Group.Item>
      <Logomark color="color-red-8" secondColor="color-red-5" width="6em" />
    </Group.Item>
  </Group>
);

export const white = () => (
  <Group>
    <Group.Item>
      <Logomark
        color="color-white"
        backgroundColor="color-grey-4"
        padding="0.75em"
        width="6em"
      />
    </Group.Item>
    <Group.Item>
      <Logomark
        color="color-white"
        backgroundColor="color-azure-4"
        padding="0.75em"
        width="6em"
      />
    </Group.Item>
    <Group.Item>
      <Logomark
        color="color-white"
        backgroundColor="color-red-8"
        padding="0.75em"
        width="6em"
      />
    </Group.Item>
  </Group>
);

const selectColor = [""].concat(Logomark.colors);
const selectSecondColor = ["", "true"].concat(Logomark.colors);

export const knobs = () => (
  <Logomark
    backgroundColor={select("backgroundColor", selectColor) || null}
    color={select("color", selectColor) || null}
    padding={text("padding", "")}
    secondColor={
      (select("secondColor", selectSecondColor) === "true"
        ? true
        : select("secondColor", selectSecondColor)) || null
    }
    title={text("title", "")}
    width={text("width", "6em")}
  />
);

knobs.story = { parameters: { docs: { disable: true } } };
