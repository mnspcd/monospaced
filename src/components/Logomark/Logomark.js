import PropTypes from "prop-types";
import React from "react";

import "./Logomark.css";
import colors, { colorKeys } from "../../system/color";

const Logomark = ({
  backgroundColor,
  color,
  padding,
  secondColor,
  title,
  width,
}) => {
  return (
    <span
      className="Logomark"
      style={{ backgroundColor: colors[backgroundColor], width }}
    >
      <span className="Logomark-inner">
        <svg
          className="Logomark-svg"
          role={title && "img"}
          style={{ padding }}
          viewBox="0 0 2048 2048"
          xmlns="http://www.w3.org/2000/svg"
          aria-hidden={!title || null}
          aria-label={title}
        >
          <path
            d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
            fill={colors[color] || "currentColor"}
          />
          {secondColor && (
            <path
              d="M767.766 1640.847h-128.51V406.846H764.35l258.78 615.283 258.8-615.283h126.822v1234h-128.54V701.67h-3.428l-253.655 604.97-251.925-604.97-3.448 3.385v935.793z"
              fill={
                typeof secondColor === "string"
                  ? colors[secondColor]
                  : colors.white
              }
            />
          )}
        </svg>
      </span>
    </span>
  );
};

Logomark.propTypes = {
  backgroundColor: PropTypes.oneOf(colorKeys),
  color: PropTypes.oneOf(colorKeys),
  padding: PropTypes.string,
  secondColor: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.oneOf(colorKeys),
  ]),
  title: PropTypes.string,
  width: PropTypes.string,
};

export default Logomark;
