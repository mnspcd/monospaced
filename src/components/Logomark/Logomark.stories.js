import React from "react";
import { select, text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Logomark from "./Logomark";
import { colorKeys } from "../../system/color";

const selectColor = [""].concat(colorKeys);
const selectSecondColor = ["", "true"].concat(colorKeys);

storiesOf("Components/Logomark", module).add("Component", () => {
  return (
    <Logomark
      backgroundColor={select("backgroundColor", selectColor) || null}
      color={select("color", selectColor) || null}
      padding={text("padding", "")}
      secondColor={
        (select("secondColor", selectSecondColor) === "true"
          ? true
          : select("secondColor", selectSecondColor)) || null
      }
      title={text("title", "")}
      width={text("width", "6em")}
    />
  );
});
