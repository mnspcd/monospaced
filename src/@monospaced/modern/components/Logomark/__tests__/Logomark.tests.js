/* eslint-env jest */
import { render } from "enzyme";
import React from "react";

import Logomark from "../Logomark";

describe("Logomark component", () => {
  it("should render correctly", () => {
    const component = render(<Logomark />);
    expect(component).toMatchSnapshot();
  });

  it("should render a title correctly", () => {
    const component = render(<Logomark title="Title" />);
    expect(component).toMatchSnapshot();
  });

  it("should render a second color correctly", () => {
    const component = render(<Logomark secondColor={"color-azure-4"} />);
    expect(component).toMatchSnapshot();
  });

  it("should render white as the default second color", () => {
    const component = render(<Logomark secondColor />);
    expect(component).toMatchSnapshot();
  });
});
