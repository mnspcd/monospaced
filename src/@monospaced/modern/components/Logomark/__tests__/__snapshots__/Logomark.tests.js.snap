// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logomark component should render a second color correctly 1`] = `
<span
  class="Logomark"
>
  <span
    class="Logomark-inner"
  >
    <svg
      aria-hidden="true"
      class="Logomark-svg"
      viewBox="0 0 2048 2048"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
        fill="currentColor"
      />
      <path
        d="M767.766 1640.847h-128.51V406.846H764.35l258.78 615.283 258.8-615.283h126.822v1234h-128.54V701.67h-3.428l-253.655 604.97-251.925-604.97-3.448 3.385v935.793z"
        fill="#00aeef"
      />
    </svg>
  </span>
</span>
`;

exports[`Logomark component should render a title correctly 1`] = `
<span
  class="Logomark"
>
  <span
    class="Logomark-inner"
  >
    <svg
      aria-label="Title"
      class="Logomark-svg"
      role="img"
      viewBox="0 0 2048 2048"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
        fill="currentColor"
      />
    </svg>
  </span>
</span>
`;

exports[`Logomark component should render correctly 1`] = `
<span
  class="Logomark"
>
  <span
    class="Logomark-inner"
  >
    <svg
      aria-hidden="true"
      class="Logomark-svg"
      viewBox="0 0 2048 2048"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
        fill="currentColor"
      />
    </svg>
  </span>
</span>
`;

exports[`Logomark component should render white as the default second color 1`] = `
<span
  class="Logomark"
>
  <span
    class="Logomark-inner"
  >
    <svg
      aria-hidden="true"
      class="Logomark-svg"
      viewBox="0 0 2048 2048"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M485.378 1921.317h1078.004v126.796H485.378zM1562.641 253.22v1541.238H485.368V253.227H1562.64zm-153.889 1387.632V406.852H1281.94l-258.81 615.277-258.78-615.277H639.257v1234.001h128.51V705.06l3.447-3.385 251.915 604.963 253.665-604.963h3.427v939.178h128.541z"
        fill="currentColor"
      />
      <path
        d="M767.766 1640.847h-128.51V406.846H764.35l258.78 615.283 258.8-615.283h126.822v1234h-128.54V701.67h-3.428l-253.655 604.97-251.925-604.97-3.448 3.385v935.793z"
        fill="#ffffff"
      />
    </svg>
  </span>
</span>
`;
